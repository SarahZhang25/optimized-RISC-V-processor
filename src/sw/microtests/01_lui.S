// This tests that your processor handles LUI correctly.
.text
start:
    lui x1, 1
    lui x2, 2
    lui x3, 3
    lui x4, 4
    lui x5, 0x7FFFF
    lui x6, 0x80000
    lui x7, 0xFFFFF
    lui x8, 0
    unimp
