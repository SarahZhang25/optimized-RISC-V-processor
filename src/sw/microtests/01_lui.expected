Dumping the state of the processor
pc = 0x00000020
x0 = 0x00000000
x1 = 0x00001000
x2 = 0x00002000
x3 = 0x00003000
x4 = 0x00004000
x5 = 0x7ffff000
x6 = 0x80000000
x7 = 0xfffff000
x8 = 0x00000000
x9 = 0x00000000
x10 = 0x00000000
x11 = 0x00000000
x12 = 0x00000000
x13 = 0x00000000
x14 = 0x00000000
x15 = 0x00000000
x16 = 0x00000000
x17 = 0x00000000
x18 = 0x00000000
x19 = 0x00000000
x20 = 0x00000000
x21 = 0x00000000
x22 = 0x00000000
x23 = 0x00000000
x24 = 0x00000000
x25 = 0x00000000
x26 = 0x00000000
x27 = 0x00000000
x28 = 0x00000000
x29 = 0x00000000
x30 = 0x00000000
x31 = 0x00000000
Quitting simulation.
